### DISCUSSION TITLE
Conditionally show columns

### DISCUSSION THREAD

**Discussion by ThisuraThejith** [Happy-Coding-Clans/vue-easytable#413] [Q&A] [answered]
I have used vue-easytable plugin version 2.8.2 to display a table inside a Vue component. I have a problem of finding how to conditionally display a table column.

In the demo displayed in [here ](https://happy-coding-clans.github.io/vue-easytable/#/en/demo)when either of the "Row Radio" or "Row Checkbox" switches are switched on we can see a column is added to the demo table dynamically. So I think this feature/functionality should be there already in the plugin but I couldn't find how to achieve this referring to the documentation.

Within my Vue component the columns array I pass to vue-easytable is as follows.

`columns: [
        {
          field: "entity",
          key: "c",
          title: "Entity",
          align: "left",
          sortBy: "asc",
        },
        {
          field: "version",
          key: "d",
          title: "Version",
          align: "center",
        },
        {
          field: "test_date",
          key: "e",
          title: "Test Date",
          align: "center",
        },
        {
          field: "score",
          key: "f",
          title: "Score",
          align: "center",
        },
        {
          field: "score_percentage",
          key: "g",
          title: "Score (%)",
          align: "center",
        },
        {
          field: "result",
          key: "h",
          title: "Result",
          align: "center",
        }
    ]`

I want to show the "Entity" column when a condition is satisfied. What should I do to achieve that?

**Comment by huangshuwei** [comment, marked as answer]
In [this example](https://happy-coding-clans.github.io/vue-easytable/#/en/demo), I implemented it through `computed`. You can refer to [this file](https://github.com/Happy-Coding-Clans/vue-easytable/blob/HEAD/examples/src/demo/index.vue) for specific code implementation .

**Reply by ThisuraThejith** [reply]
Thanks a lot @huangshuwei.

**Comment by huangshuwei** [comment]
I am also considering whether to add some APIs to support the hidden column feature

**Reply by ThisuraThejith** [reply]
It would be great if you guys could do that @huangshuwei . You guys have done a great job already by implementing vue-easytable. Keep up the great work.

**Reply by huangshuwei** [reply]
Hi, [The new version](https://github.com/Happy-Coding-Clans/vue-easytable/releases/tag/v2.11.0) adds column hidden feature.
